Add total sample counts to facets

arielsvn opened the pull request:
## Issue Number

close #1063 

## Purpose/Implementation Notes

Adds total sample counts to the filters in the API. This will require changes in the frontend as well.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

Tested endpoint locally

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots

![image](https://user-images.githubusercontent.com/1882507/54312284-e9c1e280-45ac-11e9-8e3c-8f97dac8f294.png)

Miserlou commented on api/data_refinery_api/views.py:
```diff
@@ -149,6 +149,27 @@ def get_paginated_response(self, data):
 # ElasticSearch
 ##
 from django_elasticsearch_dsl_drf.pagination import LimitOffsetPagination as ESLimitOffsetPagination
+from elasticsearch_dsl.query import Q
```
Is this actually being used?

Miserlou commented on api/data_refinery_api/views.py:
```diff
@@ -149,6 +149,27 @@ def get_paginated_response(self, data):
+
+        (Maybe there's a way to do this with the options in `ExperimentDocumentView`)
+        """
+        __facets = self.construct_facets(request, view)
```
I'd prefer it if you got rid of `__` everywhere, this is a cargo cult habit IMO..

arielsvn commented:
@Miserlou I added tests, and also modified the shape of the `facets` object to remove all ES related properties. This doesn't have any substantial impact on performance, but made it easier to tests the response of the search.

Here's an example of the `facets` object:

```
'facets': {
  'organism_names': {}, 
  'has_publication': {'false': 1}, 
  'platform_names': {}, 
  'technology': {'microarray': 1, 'rna-seq': 0}
}
```

Let me know if we can improve this further.
